tests/common/api/libraries/request_own_library.py

```diff
@@ -0,0 +1,122 @@
+import json
+import requests
+
+try:
+    from robot.libraries.BuiltIn import BuiltIn
+    from robot.libraries.BuiltIn import _Misc
+    import robot.api.logger as logger
+    from robot.api.deco import keyword
+    ROBOT = False
+except Exception:
+    ROBOT = False
+
+class request_own_library():
+    ROBOT_LIBRARY_SCOPE = 'TEST SUITE'
+    base_url = ""
+    session = requests.Session()
+    
+    @keyword('Create Session')
+    def create_session(self, base_url, headers):
+        self.base_url = base_url
+        self.session.headers.update(headers)
+
+    def do_request(self, method, endpoint, payload=None):
+        """Send REST request to the specified endpoint.
+        :param method:    str   Request method name such as GET, POST, PUT, DELETE, PATCH
+        :param endpoint:  str   Endpoint of service to which the request will be sent
+        :param payload:   dict  Data or payload send along with the REST request
+        :return: Tuple that contains the status code and the response.
+        """
+        
+        endpoint_url = f"{self.base_url}/{endpoint}"
+
+        if method in ['POST', 'PUT']:
+            response = self.session.request(method, endpoint_url, data=json.dumps(payload))
+        else:
+            response = self.session.request(method, endpoint_url)
+
+        if not response.ok:
+            return response.status_code, response.text
+
+        return response.status_code, response
+    @keyword('GET On Session')
+    def get_request(self, endpoint, expected_status_code ):
+        """Method to send GET requests to the specified endpoint
+        :param endpoint:  str   Endpoint of service to which the request will be sent
+        :param expected_status_codet:  str   Expected status code
+        :return:  the response.
+        """
+        expected = expected_status_code[len(expected_status_code)-3:]
+        
+        status_code, response = self.do_request('GET', endpoint)
+        if int(status_code) == int(expected)  :
+            return response
+        else:
+            return f"it was expected the status {expected } but the status was {status_code}"
```

Comment:
If the status code was different from the explicitly expected one then we should raise an Exception, instead of returning a string, otherwise when this is used in a test, the step will succeed even if the status code was different from the one expected.
(same for POST, PUT, DELETE)